pkg/scheduler/dependency_graph.go

```diff
@@ -381,11 +426,8 @@ func (sched *Scheduler) allocateReplicas(flow *client.Flow, options interfaces.D
 	}
 
 	sort.Stable(&tmp)
-	var result []string
-	for _, replica := range tmp.Items {
-		result = append(result, replica.ReplicaName())
-	}
-	return result[:count], nil
+
+	return tmp.Items[:count], nil, nil
 }
 
 // BuildDependencyGraph loads dependencies data and creates the DependencyGraph
```

Comment:
IMO they are correlated as all of them are part of preparation for gathering correct flow, doing it's validation.
Maybe this could be named `getCorrectFlow`?